ospfd/ospf_vty.c

```diff
@@ -5443,23 +5452,50 @@ static int show_ip_ospf_neighbor_id_common(struct vty *vty, struct ospf *ospf,
 	return CMD_SUCCESS;
 }
 
-DEFPY(show_ip_ospf_neighbor_id, show_ip_ospf_neighbor_id_cmd,
-      "show ip ospf neighbor A.B.C.D$router_id [detail$detail] [json$json]",
-      SHOW_STR IP_STR
+DEFPY(show_ip_ospf_neighbor_id,
+      show_ip_ospf_neighbor_id_cmd,
+      "show ip ospf [vrf NAME$vrf_name] neighbor A.B.C.D$router_id [detail$detail] [json$json]",
+      SHOW_STR
+      IP_STR
       "OSPF information\n"
+      VRF_CMD_HELP_STR
       "Neighbor list\n"
       "Neighbor ID\n"
-      "Detailed output\n" JSON_STR)
+      "Detailed output\n"
+      JSON_STR)
 {
 	struct ospf *ospf;
 	struct listnode *node;
 	int ret = CMD_SUCCESS;
+	int inst = 0;
 
-	for (ALL_LIST_ELEMENTS_RO(om->ospf, node, ospf)) {
-		if (!ospf->oi_running)
-			continue;
-		ret = show_ip_ospf_neighbor_id_common(vty, ospf, &router_id,
-						      !!json, 0, !!detail);
+	if (vrf_name && !strmatch(vrf_name, "all")) {
+		ospf = ospf_lookup_by_inst_name(inst, vrf_name);
+		if (ospf == NULL || !ospf->oi_running) {
+			if (!json)
+				vty_out(vty,
+					"%% OSPF is not enabled in vrf %s\n",
+					vrf_name);
+			else
+				print_empty_json_obj(vty);
+
+			return CMD_SUCCESS;
+		}
+		ret = show_ip_ospf_neighbor_id_common(
+			vty, ospf, &router_id, !!json, 0, !!detail, NULL);
+	} else {
```

Comment:
Is this OK here to do VRF stuff (JSON, etc.) even if we don't query for VRF? I mean `show ip ospf neighbor A.B.C.D` or how this is supposed to work?

Reply:
Even before I made the code change,  ```show ip ospf neighbor A.B.C.D``` was displaying the requested OSPF neighbor router ID in all the VRFs. I.e, this command was doing the equivalent of  ```show ip ospf neighbor vrf all A.B.C.D```.  Since this is the exisiting behavior, I thought it would be best not to modify it.